File: models/public_party.go
package models

import (
	"encoding/json"

	"github.com/vattle/sqlboiler/types"
)

type PublicParty struct {
	Location     types.JSON    `json:"location"`
	RoomCode     string        `json:"room_code"`
	Ended        bool          `json:"ended"`
	Guests       *types.JSON   `json:"guests,omitempty"`
	CurrentTrack *PlayingTrack `json:"current_track,omitempty"`
}

func (p *Party) Public() (PublicParty, error) {
	response := PublicParty{
		Location: p.Location,
		RoomCode: p.RoomCode,
		Ended:    p.Ended,
	}

	guests, err := p.Guests()
	if err != nil {
		return PublicParty{}, err
	}

	if len(guests) > 0 {
		guestsPublic := make([]PublicGuest, 0)
		for _, guest := range guests {
			guestsPublic = append(guestsPublic, guest.Public())
		}
		guestsJsonRaw, _ := json.Marshal(guestsPublic)
		guestsJson := types.JSON(guestsJsonRaw)
		response.Guests = &guestsJson
	}

	if p.CurrentTrack.Valid {
		var track PlayingTrack
		err := p.CurrentTrack.Unmarshal(&track)
		if err == nil {
			response.CurrentTrack = &track
		}
	}

	return response, nil
}

File: Server/Models/PublicParty.cs
using System.Collections.Generic;
using System.Linq;
using Models;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using Server.Data;

namespace Server.Models
{
  [ResourceIdentifier("party")]
  public class PublicParty
  {
    [JsonProperty("location", NullValueHandling = NullValueHandling.Include)]
    public JObject Location { get; set; }

    [JsonProperty("room_code", NullValueHandling = NullValueHandling.Include)]
    public string RoomCode { get; set; }

    [JsonProperty("ended", NullValueHandling = NullValueHandling.Include)]
    public bool Ended { get; set; }

    [JsonProperty("guests", NullValueHandling = NullValueHandling.Ignore)]
    public List<PublicGuest> Guests { get; set; }

    [JsonProperty("current_track", NullValueHandling = NullValueHandling.Ignore)]
    public PlayingTrack CurrentTrack { get; set; }

    public static PublicParty FromParty(Party party, IEnumerable<Guest> guests = null) => new PublicParty {
      Location = string.IsNullOrWhiteSpace(party.Location)
        ? null
        : JsonConvert.DeserializeObject<JObject>(party.Location),
      RoomCode = party.RoomCode,
      Ended = party.Ended,
      Guests = guests?.Select(g => new PublicGuest(g)).ToList() ?? new List<PublicGuest>(),
      CurrentTrack = string.IsNullOrWhiteSpace(party.CurrentTrack)
        ? null
        : JsonConvert.DeserializeObject<PlayingTrack>(party.CurrentTrack)
    };
  }
}
